src/python/TaskWorker/Actions/PostJob.py

```diff
@@ -130,129 +132,129 @@ def first_pj_execution():
-                    msg  = "Updating error summary for jobid %s retry %s with following information:" % (fjr_job_id, fjr_crab_retry)
-                    msg += "\n'exit code' = %s" % (exit_code)
-                    msg += "\n'exit message' = %s" % (exit_msg)
-                    logger.info(msg)
-                    error_summary.setdefault(fjr_job_id, {})[fjr_crab_retry] = (exit_code, exit_msg, {})
+                        error_summary = [exit_code, exit_msg, {}]
                     error_summary_changed = True
+        except Exception as ex:
+            logger.info(str(ex))
+            ## Write to the error summary that the job report is not valid or has no error
+            ## message
+            if not rep:
+                exit_msg = 'Invalid framework job report. The framework job report exists, but it cannot be loaded.'
+            else:
+                exit_msg = rep['exitMsg'] if 'exitMsg' in rep else 'The framework job report could be loaded, but no error message was found there.'
+            msg  = "Updating error summary for jobid %s retry %s with following information:" % (job_id, crab_retry)
+            msg += "\n'exit code' = %s" % (exit_code)
+            msg += "\n'exit message' = %s" % (exit_msg)
+            logger.info(msg)
+            error_summary = [exit_code, exit_msg, {}]
+            error_summary_changed = True
+
+    # Write the fjr report summary of this postjob to a file which task_process reads incrementally
+    if error_summary_changed and os.path.isfile("/etc/enable_task_daemon"):
+        with getLock(G_FJR_PARSE_RESULTS_FILE_NAME):
+            with open(G_FJR_PARSE_RESULTS_FILE_NAME, "a+") as fjr_parse_results:
+                fjr_parse_results.write(json.dumps({job_id : {crab_retry : error_summary}}) + "\n")
+
+    # Read, update and re-write the error_summary.json file
+    try:
+        if os.stat(G_ERROR_SUMMARY_FILE_NAME).st_size != 0:
+            fsummary.seek(0)
+            error_summary_old_content = json.load(fsummary)
+        else:
+            error_summary_old_content = {}
+    except (IOError, ValueError):
+        ## There is nothing to do if the error_summary file doesn't exist or is invalid.
+        ## Just recreate it.
+        logger.info("File %s is empty, wrong or does not exist. Will create a new file." % (G_ERROR_SUMMARY_FILE_NAME))
+    error_summary_new_content = error_summary_old_content
+    error_summary_new_content[job_id] = {crab_retry : error_summary}
 
     logger.debug("Writing error summary file")
     ## If we have updated the error summary, write it to the json file.
     ## Use a temporary file and rename to avoid concurrent writing of the file.
     if error_summary_changed:
-        fsummary.truncate(0)
-        json.dump(error_summary, fsummary)
+        fsummary.seek(0)
+        fsummary.truncate()
```

Comment:
truncate(0) sets the file size to zero, it does not change the cursor position. seek(0) +  truncate() sets the cursor position to the beginning of the file and then truncates the file to that position (same as seek(0) + truncate(0), I should add a zero to the truncate function to make it more clear).

[1] https://docs.python.org/2.6/library/stdtypes.html#file.truncate
[2] http://stackoverflow.com/questions/31624747/if-i-truncate-a-file-to-zero-in-python-3-do-i-also-need-to-seek-to-position-zero